### PR TITLE
Replace images in wxRendererMac::DrawTitleBarBitmap() with drawing code.

### PR DESCRIPTION
**Pull request by TcT2k**
The low resolution (14x14) bitmaps scaled badly on high resolution displays. A close button suitable for usage inside a window (like `wxInfoBar`) is not available via HI theme drawing methods. This drawing code tries to emulate a close button, as close as possible to the one found in the Xcode 6+ welcome window.

Normal: screen shot 2015-09-25 at 15 20 33 Hover: screen shot 2015-09-25 at 15 20 41 Old implementation: screen shot 2015-09-25 at 15 43 48

### CODE REVIEW

**Comment by TcT2k** [comment]
@vadz previous comments where lost when I pushed the updated implementation, so I'll reply here:
I wasn't even ware that `GetContentScaleFactor()` returns something on MSW. I've updated the second commit to include other ports too. It looks much better now on Windows. But a little bit of platform specific code is still required as only the OSX version of `wxBitmap` has ctors with a specifiable content scaling factor.

Windows 8.1 (200%):
wxwidgets_closebtn_win

File: src/osx/carbon/renderer.cpp [hunk 1]
```diff
@@ -699,218 +699,55 @@ void wxRendererMac::DrawTitleBarBitmap(wxWindow *win,
...
+        wxRect circleRect(rect);
+        circleRect.Inflate(-2 * contentScale);
+    
+        dc.DrawEllipse(circleRect);
+    }
+    
+    dc.SetPen(wxPen(glyphColor, 1 * contentScale));
+    
+    wxRect centerRect(rect);
+    centerRect.Inflate(-5 * contentScale);
```
Comment (vadz) [inline src/osx/carbon/renderer.cpp]: This could be written a bit more clearly as `Deflate(5*contentScale)`.

File: src/osx/carbon/renderer.cpp [hunk 2]
```diff
@@ -699,218 +699,55 @@ void wxRendererMac::DrawTitleBarBitmap(wxWindow *win,
...
+    
+    wxColour glyphColor;
+    
+    bool drawCircle;
     if ( flags & wxCONTROL_PRESSED )
     {
-        data = close_pressed_png;
-        len = WXSIZEOF(close_pressed_png);
+        drawCircle = true;
+        glyphColor = wxColour(104, 104, 104);
```
Comment (vadz) [inline src/osx/carbon/renderer.cpp]: It would be nice to know where do all these hardcoded RGB values come from.

File: src/common/bmpbtncmn.cpp [hunk 3]
```diff
@@ -103,13 +103,21 @@ GetCloseButtonBitmap(wxWindow *win,
...
+    wxBitmap bmp(drawSize);
     wxMemoryDC dc(bmp);
     dc.SetBackground(colBg);
     dc.Clear();
+
+
     wxRendererNative::Get().
-        DrawTitleBarBitmap(win, dc, size, wxTITLEBAR_BUTTON_CLOSE, flags);
+        DrawTitleBarBitmap(win, dc, drawSize, wxTITLEBAR_BUTTON_CLOSE, flags);
+#ifdef __WXOSX__
```
Comment (vadz) [inline src/common/bmpbtncmn.cpp]: Wouldn't using `wxBitmap::CreateScaled()` make it possible to do this without any `#ifdefs`?

**Comment by vadz** [comment]
Sorry for the delay, I've finally reviewed the latest version of the patch. If the latest comments could be addressed (especially the one about `CreateScaled()`), I promise to apply this without any further delay. TIA!

**Comment by TcT2k** [comment]
I've integrated your comments, but I wasn't able to use `wxBitmap::CreateScaled()` (which is currently undocumented) as the result was an empty bitmap. I'm not really sure why as the code looks correct when stepping into it. It would obviously be cleaner without the #ifdef but I couldn't find a way to make it work without the cocoa specific ctor.

**Comment by vadz** [comment]
It would be great if somebody with a retina Mac could debug this, it's really appalling that we don't have a way to create a bitmap in a portable way :-(

Also, sorry to be a pain, but what about the hardcoded RGB values? They seem to be there without any explanations...

**Comment by TcT2k** [comment]
@vadz 
I've misinterpreted `CreateScaled`. I thought it would create a scaled bitmap based and the current content, but obviously it creates the bitmap object like other `Create` methods.
The code now works without ifdefs. I've tested OS X and Win 8.1 with a 4K display and the results match the screenshots posted above.

As for the RGB values I've included a comment that they are hard coded based on the Xcode 6+ welcome screen. I haven't found a good match using any system colors, which would obviously be a nicer approach.

I've also added documentation for `CreateScaled`in an additional commit (3ab3acd).

**Comment by vadz** [comment]
Great, thanks for fixing this!

**Pull request merged**